homework/minieieva_daria/report.md

```diff
@@ -0,0 +1,100 @@
+### Week 6 Regular Expressions Homework
+
+### This is a template for the report of 21 problems on [this site](http://regextutorials.com/excercise.html).
+
+### Fork this repository, change this file and send us the link for grading.
+
+### We recommend you (re)watch our presentation, and use a [cheatsheet](./cheatsheet.md) while working on the following problems.
+
+### If you are having problems with figuring out the correct solution, skip it for later, and if you are completely lost, use the hints and try to understand them.
+
+---
+
+### Problem 1: Floating point numbers
+```
+\d\.\d+
+```
+### Problem 2: Years before 1990
+```
+\d\s\w*\s\w*[,|:]*(\s\w*)*\(1[0-9][0-8][0-9]\)
```

Comment:
I am fairly sure this can be done a lot easier.

For example, there is no need for `[,|:]` the pattern works just as well.

This works just as well: `\d(\s\w*)*\(1[0-9][0-8][0-9]\)`, and it's far from ideal.